src/eckit/geo/Domain.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+
+namespace eckit::geo {
+
+
+class Domain {
+public:
+    // -- Exceptions
+    // None
+
+    // -- Constructors
+
+    Domain(double north, double west, double south, double east);
+    Domain();
+    Domain(const Domain&) = default;
+    Domain(Domain&&)      = default;
+
+    // -- Destructor
+
+    virtual ~Domain() = default;
+
+    // -- Convertors
+    // None
+
+    // -- Operators
+
+    Domain& operator=(const Domain&) = default;
+    Domain& operator=(Domain&&)      = default;
+    bool operator==(const Domain&) const;
+    bool operator!=(const Domain& other) const { return !operator==(other); }
+
+    // -- Methods
+
+    double north() const { return north_; }
+    double west() const { return west_; }
+    double south() const { return south_; }
+    double east() const { return east_; }
+
+    bool includesNorthPole() const;
+    bool includesSouthPole() const;
+    bool isPeriodicWestEast() const;
+
+    bool contains(double lat, double lon) const;
+    bool contains(const Domain&) const;
+    bool empty() const;
+    double area(double radius) const;
```

Comment:
What does this do?

Reply:
This class is supposed to be a drop in replacement for mir's usage of the coverage of the grid area, as limited by 2 parallels and 2 meridians, and with double as a coordinate type (as distinguished from the mir's BoundingBox). But, it is overlapping in concept with the eckit::geo::BoundingBox, and this will be updated with a more general base class "Area" (the discussion hasn't yet taken place).

So I'm removing this now to make it more clear

Reply:
I was wondering what `double area(radius)` is supposed to return. A circular area around some kind of centrepoint?

Reply:
I've removed the class not to be a problem (it isn't really needed for eckit::geo, so it's a good removal!), but I have the same methods for eckit::geo::BoundingBox. There "double area(radius)" means the surface area represented by the bounding box on a sphere of given raidus. I see that it should instead be part of Figure (which accounts for that supporting geometry), jult like a great-circle distance and the like